src/visual.ts

```diff
@@ -1278,6 +1278,82 @@ module powerbi.extensibility.visual {
                 .exit()
                 .remove();
 
+            // create labels on link as A - B : Value
+            let linkTextData: SankeyDiagramLink[] = sankeyDiagramDataView.links.filter((link: SankeyDiagramLink) => {
+                return link.height > SankeyDiagram.MinSize && this.dataView.settings.linkLabels.show;
+            });
+            // add defs element to svg
+            let svgDefs = this.root
+                .selectAll("defs");
+
+            let svgDefsSelection = svgDefs.data([1]);
+            svgDefsSelection
+                .enter()
+                .append("defs");
+
+            svgDefsSelection
+                .exit()
+                .remove();
+
+            let singleDefsElement = d3.select(svgDefsSelection.node());
+
+            // add text path for lables
+            let linkLabelPaths = singleDefsElement.selectAll(".linkLabelPaths");
```

Comment:
Can we create class name using ClassAndSelector class?
You might take a look at SankeyDiagram.LinkSelector as an example.